### DISCUSSION TITLE
Query behaves differently after upgrade to v0.100.0

### DISCUSSION THREAD

**Discussion by gabriel-frattini** [fastapi/fastapi#9895] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Query


FOO = Query()


from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def root(foo: str = FOO, bar: str = FOO):
    return {"foo": foo, "bar": bar}
```

### Description

* Call the endpoint with foo=foo&bar=bar as parameters
* With v0.98.0 It returns a JSON blob with
```
{
    "foo": "foo",
    "bar": "bar"
}
```

* With v0.100.0 It returns a JSON blob with 
```
{
    "foo": "foo",
    "bar": "foo"
}
```

* What is the expected behaviour here?
Did something change in v0.100.0 so `foo` and `bar` now references the same instance FOO?

### Operating System

macOS

### FastAPI Version

0.100.0

### Python Version

3.11.0

**Comment by Kludex** [comment, marked as answer]
This is a bug. Thanks. 👍
